test/Integration.t.sol

```diff
@@ -477,4 +477,106 @@ contract ConcreteTests is CapAutomatorIntegrationTestsBase {
         vm.stopPrank();
     }
 
+    function test_E2E_flashloan() public {
+        DataTypes.ReserveData memory initialReserveData = pool.getReserveData(WBTC);
+
+        // Confirm initial state
+        assertEq(ERC20(WBTC).decimals(), 8);
+        assertEq(currentATokenSupply(initialReserveData), 750);
+        assertEq(initialReserveData.configuration.getSupplyCap(), 3_000);
+
+        vm.prank(SPARK_PROXY);
+        capAutomator.setSupplyCapConfig({
+            asset:            WBTC,
+            max:              10_000,
+            gap:              50,
+            increaseCooldown: 12 hours
+        });
+
+        capAutomator.execSupply(WBTC);
+        vm.roll(block.number + 1);
+        skip(12 hours + 1);
+
+        // Confirm new, tightly set supply cap
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 800);
+
+        pool.flashLoanSimple(
+            address(this),
+            WBTC,
+            100e8,
+            abi.encode(address(pool)),
+            0
+        );
+
+        // Confirm the cap increase even though the supply effectively didn't change
+        DataTypes.ReserveData memory postFlashloanReserveData = pool.getReserveData(WBTC);
+        assertEq(currentATokenSupply(postFlashloanReserveData), 750);
+        assertEq(postFlashloanReserveData.configuration.getSupplyCap(), 840);
+
+        // Confirm that in the next block, before increase cooldown passes, cap can be decreased
+        vm.roll(block.number + 1);
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 800);
+
+        vm.roll(block.number + 1);
+        skip(6 hours);
+
+        deal(WBTC, address(this), 40e8);
+        ERC20(WBTC).approve(address(pool), 40e8);
+        pool.supply(WBTC, 40e8, address(this), 0);
+
+        // Confirm that after some time, smaller than cooldwon, cap cannot be increased
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 800);
+
+        skip(6 hours + 1);
+
+        // Confirm that after suffcient time passes, cap can be increased again
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 840);
+    }
+
+    // Called back from the flashloan
+    function executeOperation(
+        address asset,
+        uint256 amount,
+        uint256,
+        address,
+        bytes calldata
+    ) external returns (bool) {
+        // Supply additional funds to the pool, bring the supply closer to the cap than the gap
+        ERC20(asset).approve(address(pool), 40e8);
+        pool.supply(WBTC, 40e8, address(this), 0);
```

Comment:
For this example I think we should supply the max amount to demo the max increase possible with this function

Reply:
My understanding is its cap + gap but we should show that

Reply:
In order to do that, I'd need to `deal` some more tokens to `address(this)` to have enough to supply to full and also the numbers would be much uglier and less round. I think this test tests the logic it should test, while edge cases are tested elsewhere. But if you have a strong opinion here, I can add these changes.